features/mbedtls/platform/inc/mbedtls_config_reader.h

```diff
@@ -410,38 +411,46 @@
 #define MBEDTLS_SHA256_C
 #define MBEDTLS_HAVE_TIME
 #define MBEDTLS_REMOVE_ARC4_CIPHERSUITES
-#define MBEDTLS_SSL_TLS_C
 #define MBEDTLS_CIPHER_C
 #define MBEDTLS_MD_C
 #define MBEDTLS_ERROR_STRERROR_DUMMY
-#define MBEDTLS_NO_PLATFORM_ENTROPY
-#define MBEDTLS_SSL_EXTENDED_MASTER_SECRET
-#define MBEDTLS_SSL_PROTO_TLS1_2
 #define MBEDTLS_X509_CHECK_KEY_USAGE
 #define MBEDTLS_X509_CHECK_EXTENDED_KEY_USAGE
 #define MBEDTLS_VERSION_C
 #define MBEDTLS_SELF_TEST
 #define MBEDTLS_PK_RSA_ALT_SUPPORT
-#define MBEDTLS_SSL_ALL_ALERT_MESSAGES
-#define MBEDTLS_SSL_ENCRYPT_THEN_MAC
-#define MBEDTLS_SSL_RENEGOTIATION
-#define MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
-#define MBEDTLS_SSL_PROTO_DTLS
-#define MBEDTLS_SSL_ALPN
-#define MBEDTLS_SSL_DTLS_ANTI_REPLAY
-#define MBEDTLS_SSL_DTLS_HELLO_VERIFY
-#define MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
-#define MBEDTLS_SSL_DTLS_BADMAC_LIMIT
 #define MBEDTLS_SSL_SESSION_TICKETS
-#define MBEDTLS_SSL_EXPORT_KEYS
 #define MBEDTLS_VERSION_FEATURES
 #define MBEDTLS_CTR_DRBG_C
```

Comment:
Shouldn't this be defined only if entropy is defined? (e.g. in https://github.com/ARMmbed/mbed-os/pull/9669/commits/e649409bf7039f9378cc9475160d32193e5e9277#diff-bcd8e25250210214920a5f19424c5709R432)

Reply:
agreed